boards/BMS/Inc/App/App_Imd.h

```diff
@@ -2,21 +2,55 @@
 
 struct Imd;
 
-#include "auto_generated/App_CanTx.h"
+// States for speed start measurement
+enum SST
+{
+    SST_GOOD,
+    SST_BAD,
+    SST_INVALID
+};
+
+// The IMD encodes condition in its PWM output's frequency
+enum Imd_Condition
+{
+    IMD_SHORT_CIRCUIT,
+    IMD_NORMAL,
+    IMD_UNDERVOLTAGE_DETECTED,
+    IMD_SST,
+    IMD_DEVICE_ERROR,
+    IMD_EARTH_FAULT,
+    NUM_OF_IMD_CONDITIONS,
+    IMD_INVALID = NUM_OF_IMD_CONDITIONS,
+};
+
+struct Imd_PwmEncoding
+{
+    bool valid_duty_cycle;
+    union {
+        // 0Hz, 40Hz, 50Hz: PWM doesn't encode any information
+        uint8_t dummy;
+        // 10 and 20Hz: Insulation measurement DCP
+        uint16_t insulation_measurement_dcp_kohms;
+        // 30Hz: Speed Start Measurement
+        enum SST speed_start_status;
+    };
+};
 
 /**
  * Allocate and initialize an IMD
  * @get_pwm_frequency: A function that can be called to get the frequency of the
  *                     IMD's PWM output
+ * @get_pwm_frequency_tolerance: A function that can be called to get the
+ *                               frequency tolerance of the IMD's PWM output
  * @get_pwm_duty_cycle: A function that can be called to get the duty cycle of
  *                      the IMD's PWM output
  * @get_seconds_since_power_on: A function that can be called to get the time of
  *                              the system, in milliseconds
  * @return A pointer to the created IMD, whose ownership is given to the caller
  */
 struct Imd *App_Imd_Create(
-    struct BmsCanTxInterface *can_tx,
     float (*get_pwm_frequency)(void),
+    float (*get_pwm_frequency_tolerance)(void),
```

Comment:
Transferring the unresolved comment from https://github.com/UBCFormulaElectric/Consolidated-Firmware/pull/536#discussion_r405608157